website/src/components/misc/article-sections.tsx

```diff
@@ -14,86 +14,99 @@ export const ArticleSections: FunctionComponent<ArticleSectionsProperties> = ({
   data,
 }) => {
   const dispatch = useDispatch();
+  const [activeHeadingId, setActiveHeadingId] = useState<string | undefined>();
 
   const handleCloseClick = useCallback(() => {
     dispatch(closeAside());
   }, []);
 
   useEffect(() => {
-    const ids = data
-      .tableOfContents!.split(/"|\//)
-      .filter((item) => item.indexOf("#") === 0)
-      .map((item) => {
-        const id = item.substring(1);
-
-        return {
-          id: `toc-${id}`,
-          position: document.getElementById(id)!.offsetTop - 80,
-        };
-      })
-      .reverse();
-    let currentActiveId: string | undefined;
-
-    const handler = () => {
+    const headingPositions =
+      ((data.tableOfContents.items as TableOfContentItem[]) ?? [])
+        .flatMap(item => ([item, ...item.items ?? []]))
+        .map(item => {
+          const headingPosition = document.getElementById(item.url.substring(1))!
```

Comment:
This could be named `position`.

Reply:
Changed in 6722f3cf533d023def9e090f50c330edfc9296dc